samples/semihosting/include/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+# Generate an array with random data.
+set(RAW_PIXELS_SIZE 102400)
+set(RAW_PIXELS_HEADER_PATH ${CMAKE_CURRENT_BINARY_DIR}/raw_pixels.hpp)
+file(WRITE ${RAW_PIXELS_HEADER_PATH})
+file(APPEND ${RAW_PIXELS_HEADER_PATH} "#ifndef RAW_PIXELS_HPP\n")
+file(APPEND ${RAW_PIXELS_HEADER_PATH} "#define RAW_PIXELS_HPP\n")
+file(APPEND ${RAW_PIXELS_HEADER_PATH} "#include <cstdint>\n")
+file(APPEND ${RAW_PIXELS_HEADER_PATH} "#define RAW_PIXELS_SIZE ${RAW_PIXELS_SIZE}\n")
+
+# Seed the random number generator.
+string(RANDOM LENGTH 8 ALPHABET 0123456789abcdf RANDOM_SEED 314 number)
+
+file(APPEND ${RAW_PIXELS_HEADER_PATH} "static std::uint32_t raw_pixels[RAW_PIXELS_SIZE + 1] = {\n")
+foreach(i RANGE ${RAW_PIXELS_SIZE})
+  string(RANDOM LENGTH 8 ALPHABET 0123456789abcdf number)
+  file(APPEND ${RAW_PIXELS_HEADER_PATH} "0x${number},\n")
+endforeach()
+file(APPEND ${RAW_PIXELS_HEADER_PATH} "};\n")
+file(APPEND ${RAW_PIXELS_HEADER_PATH} "#endif //RAW_PIXELS_HPP\n")
```

Comment:
I propose to use template and fill options/random values. https://cmake.org/cmake/help/latest/command/configure_file.html. It's readable and easy to modify.

Reply:
Just to make sure I understand:

1. Have a header file that has the setting of the variable, something like:
```
#ifndef RAW_PIXELS_HPP
#define RAW_PIXELS_HPP
static std::uint32_t raw_pixels[RAW_PIXELS_SIZE + 1] = {
__SOME_MACRO
}; 
#endif //RAW_PIXELS_HPP
```

2. Use the `string(RANDOM LENGTH ...` facility to populate a variable, say `RANDOM_PIXELS` in cmake with all the random variables

3. Use https://cmake.org/cmake/help/latest/command/configure_file.html to write `RANDOM_PIXEL` into the header file.

I like the idea (I agree it will make it easier to modify), I am just making sure my understanding is correct.